iml-api/src/graphql.rs

```diff
@@ -555,3 +618,48 @@ async fn get_fs_target_resources(
 
     Ok(xs)
 }
+
+async fn get_banned_targets(pool: &PgPool) -> Result<Vec<BannedTargetResource>, ImlApiError> {
+    let xs = sqlx::query!(r#"
+            SELECT b.id, b.resource, b.node, b.cluster_id, nh.host_id, t.mount_point
+            FROM corosync_resource_bans b
+            INNER JOIN corosync_node_managed_host nh ON (nh.corosync_node_id).name = b.node
+            AND nh.cluster_id = b.cluster_id
+            INNER JOIN corosync_target_resource t ON t.id = b.resource AND b.cluster_id = t.cluster_id
+        "#)
+        .fetch(pool)
+        .map_ok(|x| {
+            BannedTargetResource {
+                resource: x.resource,
+                cluster_id: x.cluster_id,
+                host_id: x.host_id,
+                mount_point: x.mount_point,
+            }
+        })
+        .try_collect()
+        .await?;
+
+    Ok(xs)
+}
+
+async fn get_banned_resources(pool: &PgPool) -> Result<Vec<BannedResource>, ImlApiError> {
+    let xs = sqlx::query!(
+        r#"
+            SELECT id, cluster_id, resource, node, weight, master_only
+            FROM corosync_resource_bans
+        "#
+    )
+    .fetch(pool)
+    .map_ok(|x| BannedResource {
+        id: x.id,
+        cluster_id: x.cluster_id,
+        resource: x.resource,
+        node: x.node,
+        weight: x.weight,
+        master_only: x.master_only,
+    })
+    .try_collect()
+    .await?;
```

Comment:
This would be better expressed as something like:

```suggestion
    let xs = sqlx::query_as!(BannedResource,
        r#"
            SELECT id, cluster_id, resource, node, weight, master_only
            FROM corosync_resource_bans
        "#
    )
    .fetch_all(pool)
    .await?;
```

Reply:
sure, I'll update.